### PR TITLE
block: Adds Download and MarkForNoCompact

### PR DESCRIPTION
**Pull request by cyriltovena**
Download basically sync a dir locally and MarkForNoCompact allow to make a block that should not be compacted.

I've also added a bunch of missing tests.

### CODE REVIEW

File: pkg/phlaredb/block/block.go [hunk 1]
```diff
@@ -52,6 +52,26 @@ func DownloadMeta(ctx context.Context, logger log.Logger, bkt objstore.Bucket, i
 	return m, nil
 }
 
+// Download downloads directory that is meant to be block directory. If any of the files
+// have a hash calculated in the meta file and it matches with what is in the destination path then
+// we do not download it. We always re-download the meta file.
+func Download(ctx context.Context, logger log.Logger, bucket objstore.Bucket, id ulid.ULID, dst string, options ...objstore.DownloadOption) error {
```
Comment (kolesnikovae) [inline pkg/phlaredb/block/block.go:58]: A nit: just to be clear as we are not actually calculating hashes

```suggestion
// Download downloads directory that is meant to be block directory.
func Download(ctx context.Context, logger log.Logger, bucket objstore.Bucket, id ulid.ULID, dst string, options ...objstore.DownloadOption) error {
```

File: pkg/util/labels.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"github.com/cespare/xxhash/v2"
+	"github.com/prometheus/prometheus/model/labels"
+)
+
+var seps = []byte{'\xff'}
+
+// StableHash is a labels hashing implementation which is guaranteed to not change over time.
+// This function should be used whenever labels hashing backward compatibility must be guaranteed.
+func StableHash(ls labels.Labels) uint64 {
```
Comment (kolesnikovae) [inline pkg/util/labels.go:12]: Can't find use of the function. If we need it, should we maybe place it in the `model` package?

**Review by kolesnikovae — APPROVED** [review]
LGTM! Left a few minor notes

File: pkg/util/labels.go (hunk 2 above)
Comment (cyriltovena) [reply pkg/util/labels.go:12]: We'll use it for the compactor yes ! I can move it to model.

**Pull request merged**